includes/helpers/functions/general.php

```diff
@@ -93,14 +93,38 @@ function wpbdp_get_page_ids_with_query( $page_id ) {
 
 	$q .= ' ORDER BY ID DESC ';
 
-	return WPBDP_Utils::check_cache(
+	if ( ! wp_using_ext_object_cache() ) {
+		$results = WPBDP_Utils::check_transient_cache(
+			array(
+				'cache_key' => 'wpbdp_page_ids' . $page_id,
+				'group'     => 'wpbdp_pages',
```

Comment:
I'm thinking we should either simplify this a lot, or send 'query' and 'type' in this array so it can all be handled in one place.

The simple way:
```
if ( ! wp_using_ext_object_cache() ) {
$results = get_transient( 'wpbdp_page_ids' . $page_id );
if ( $results !== false ) {
  return $results;
}
}

$results = WPBDP_Utils::check_cache( ... );
if ( ! wp_using_ext_object_cache() ) {
set_transient( 'wpbdp_page_ids' . $page_id,  $results );
}
```

Reply:
I think having it all as an array in an option would be easier as it makes it easier to clear the cache

Reply:
If we use an option, I think we do want it to autoload since this is a query that runs on every page. This option solution still has the issue of possibly never refreshing on option that isn't called often. If we do it this way, the expiration needs to be saved for each option individually.

There are only 4 possible $page_id parameters here, so clearing the cache isn't complicated.
https://github.com/Strategy11/business-directory-plugin/blob/78e6ac7a1a9bcf2af1913b90e6f711b54d7cfded/includes/helpers/functions/general.php#L25-L29

So, we still need to either do this the simple way, or handle these three functions from the WPBDP_Utils class, by sending 'query' and 'type' to the array.

With all the complication this adds, I would vote the simple option that will be much less likely to introduce bugs.